softlayer/data_source_softlayer_dns_domain.go

```diff
@@ -13,7 +13,7 @@ func dataSourceSoftLayerDnsDomain() *schema.Resource {
 		Read: dataSourceSoftLayerDnsDomainRead,
 
 		Schema: map[string]*schema.Schema{
-			"id": &schema.Schema{
```

Comment:
A bit confused about this still. Do we know if Terraform simply _owns_ the resource id, generating its own internal one, or is the intent for providers to never have to publish an id as part of its resource schema, but the expectation is still that the provider decides what the id should be? Maybe a question for the terraform mailing list?

If Terraform still expects the provider to decide what the resource id will be, but just does not want the id as part of the schema, then I understand now and we just need to remove the ids from the schema.

Reply:
my understanding is Terraform will manage the resource id. That's why I created the first PR just remove them from the plugin.

Reply:
@c4po Sorry for my misunderstanding of this. I [asked on the Terraform mailing](https://groups.google.com/forum/#!topic/terraform-tool/8RSiwR86z60) list to make sure.

Terraform core actually does not manage the resource id. It is simply an implicit property. The provider will still assign the id to the resource. So the only change required is what you had done initially, which was to remove the id from the schema.

Reply:
ok, do you want me close the PR and work on the previous one?

Reply:
Yes, what you had initially was enough (sorry :-/)

Reply:
@renier is it ok to merge this and have a release